### PR TITLE
Cleanups and version bumps

### PR DESCRIPTION
**Pull request by xtrm-en**
Cleans-up some warnings, bumps Gradle to 8.9 and Kotlin to 2.0.0 (everywhere), and removes unused files.

This also adds an `apiJar` task and a `-api` jar artifact & publlication.

### CODE REVIEW

File: gradle-plugin/src/main/kotlin/net/weavemc/gradle/configuration/DependencyManager.kt [hunk 1]
```diff
@@ -35,10 +36,7 @@ private fun Project.addMinecraftAssets(version: MinecraftVersion) {
     val client = versionInfo.downloads.client
     DownloadUtil.checksumAndDownload(URL(client.url), client.sha1, version.minecraftJarCache.toPath())
 
-    repositories.maven {
-        name = "mojang"
-        setUrl("https://libraries.minecraft.net/")
-    }
+    repositories.maven("https://libraries.minecraft.net/")
```
Comment (xtrm-en) [inline gradle-plugin/src/main/kotlin/net/weavemc/gradle/configuration/DependencyManager.kt:39]: The Kotlin DSL simply refused to work here, so this was my solution, if you're able to make it work lmk

File: build-logic/build.gradle.kts [hunk 2]
```diff
@@ -1,5 +1,7 @@
+import org.jetbrains.kotlin.gradle.dsl.KotlinVersion
+
 plugins {
-    `kotlin-dsl`
+    id("org.gradle.kotlin.kotlin-dsl") version "4.4.0"
```
Comment (770grappenmaker) [inline build-logic/build.gradle.kts:4]: I'm assuming this is for Kotlin 2.0 support right?

File: gradle-plugin/src/main/kotlin/net/weavemc/gradle/configuration/DependencyManager.kt (hunk 1 above)
Comment (770grappenmaker) [reply gradle-plugin/src/main/kotlin/net/weavemc/gradle/configuration/DependencyManager.kt:39]: This is fine.

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.8-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.9-bin.zip
```
Comment (770grappenmaker) [inline gradle/wrapper/gradle-wrapper.properties:3]: Could you run `./gradlew wrapper`, such that `gradlew` and `gradlew.bat` etc. are updated correctly?

**Review by 770grappenmaker — COMMENTED** [review]
Not so sure I am onboard with `apiJar`, rest looks good

File: build-logic/build.gradle.kts (hunk 2 above)
Comment (xtrm-en) [reply build-logic/build.gradle.kts:4]: No, this is just a greater version of the plugin, but 2.0.0 isn't provided yet, this is leftover code from testing it

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (xtrm-en) [reply gradle/wrapper/gradle-wrapper.properties:3]: Shell scripts weren't updated, but I guess i can update the wrapper jar prematurely

**Comment by 770grappenmaker** [comment]
I like it, picasso

**Pull request merged**